webhooks/pod_webhook.go

```diff
@@ -82,40 +84,56 @@ func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admissio
 		return admission.Denied(err.Error())
 	}
 
-	// Check to see whether the FeatureFlagConfiguration has service or sync overrides
-	ff := m.getFeatureFlag(ctx, val, req.Namespace)
-
-	if ff.Spec.SyncProvider != nil && !ff.Spec.SyncProvider.IsKubernetes() {
-		// Check for ConfigMap and create it if it doesn't exist (only required if sync provider isn't kubernetes)
-		cm := corev1.ConfigMap{}
-		if err := m.Client.Get(ctx, client.ObjectKey{Name: val, Namespace: req.Namespace}, &cm); errors.IsNotFound(err) {
-			err := m.createConfigMap(ctx, val, req.Namespace, pod)
-			if err != nil {
-				m.Log.V(1).Info(fmt.Sprintf("failed to create config map %s error: %s", val, err.Error()))
-				return admission.Errored(http.StatusInternalServerError, err)
-			}
+	ffConfigs := []*corev1alpha1.FeatureFlagConfiguration{}
+	for _, ffName := range ffNames {
+		ns, name := parseAnnotation(ffName, req.Namespace)
+		if err != nil {
+			m.Log.V(1).Info(fmt.Sprintf("failed to parse annotation %s error: %s", ffName, err.Error()))
+			return admission.Errored(http.StatusBadRequest, err)
 		}
+		// Check to see whether the FeatureFlagConfiguration has service or sync overrides
+		ff := m.getFeatureFlag(ctx, name, ns)
+		if ff.Spec.SyncProvider != nil && !ff.Spec.SyncProvider.IsKubernetes() {
```

Comment:
Here we are saying that if the provider is filepath or http we should create the configmap; does http need it?

Reply:
updated this to check for file path sync, rather than non-kubernetes sync